pkg/infra/baseagent.go

```diff
@@ -222,5 +249,19 @@ func (a *Base) HandleResponse(msg messages.BoolResponseMessage)
 func (a *Base) HandleStateFoodTaken(msg messages.StateFoodTakenMessage)                  {}
 func (a *Base) HandleStateHP(msg messages.StateHPMessage)                                {}
 func (a *Base) HandleStateIntendedFoodTaken(msg messages.StateIntendedFoodIntakeMessage) {}
-func (a *Base) HandleProposeTreaty(msg messages.ProposeTreatyMessage)                    {}
-func (a *Base) HandleTreatyResponse(msg messages.TreatyResponseMessage)                  {}
+
+func (a *Base) HandleProposeTreaty(msg messages.ProposeTreatyMessage) {
+	treatyProposal := msg.Treaty()
+	// always add treaty to active list if proposing
+	a.AddToActiveTreaties(treatyProposal)
+}
+
```

Comment:
I don't think this is correct, `HandleProposeTreaty` is called on the agent that receives a treaty proposal, not the agent who sent one. The handler needs to be over-writable by agents. I suggest leaving it blank and making sure later that agent teams call `AddToActiveTreaties` correctly.

Reply:
The only issue is that this is something that needs to happen identically for all agents (treaties must be pushed to memory, what happens after is the agent team's call), so should really be a base function. Process treaties should be a function that's overwrite-able